src/gui/macOS/fileproviderdomainmanager_mac.mm

```diff
@@ -29,41 +31,98 @@
 // are consistent throughout these classes
 namespace {
 
+static constexpr auto bundleExtensions = std::array{
+    QLatin1StringView(".app"),
+    QLatin1StringView(".framework"),
+    QLatin1StringView(".kext"),
+    QLatin1StringView(".plugin"),
+    QLatin1StringView(".docset"),
+    QLatin1StringView(".xpc"),
+    QLatin1StringView(".qlgenerator"),
+    QLatin1StringView(".component"),
+    QLatin1StringView(".saver"),
+    QLatin1StringView(".mdimporter")
+};
+static const QRegularExpression illegalChars("[:/]");
+
+inline bool hasBundleExtension(const QString &domainId)
+{
+    return std::any_of(bundleExtensions.begin(), bundleExtensions.end(), [&domainId](const auto &ext) {
+        return domainId.endsWith(ext);
+    });
+}
+
+inline bool illegalDomainIdentifier(const QString &domainId)
+{
+    return !domainId.isEmpty() && !illegalChars.match(domainId).hasMatch() && hasBundleExtension(domainId);
+}
+
 QString domainIdentifierForAccount(const OCC::Account * const account)
 {
     Q_ASSERT(account);
-    static const QRegularExpression illegalChars("[:/]");
-    return account->userIdAtHostWithPort().replace(illegalChars, "-");
+    auto domainId = account->userIdAtHostWithPort();
+    Q_ASSERT(!domainId.isEmpty());
+
+    domainId.replace(illegalChars, "-");
+
+    // Some url domains like .app cause issues on macOS as these are also bundle extensions.
+    // Under the hood, fileproviderd will create a folder for the user to access the files named
+    // after the domain identifier. If the url domain is the same as a bundle extension, Finder
+    // will interpret this folder as a bundle and will not allow the user to access the files.
+    // Here we wrap the dot in the url domain extension to prevent this from happening.
+    for (const auto &ext : bundleExtensions) {
+        if (domainId.endsWith(ext)) {
```

Comment:
this will work only as long as there is no port other than 80 or 443 specified, as `userIdAtHostWithPort` also contains the port in such a case

Reply:
Argh, you are right. I will rethink this

Reply:
Actually, that's ok -- that won't cause an issue with Finder

We have issues when the folder name ends in `.app`, for instance, but if there is a port specified that means the folder name won't end in a bundle extension

Reply:
ah, true :)  all good then